src/yb/master/yql_partitions_vtable.h

```diff
@@ -32,9 +33,9 @@ class YQLPartitionsVTable : public YQLVirtualTable {
   Schema CreateSchema() const;
 
   mutable boost::shared_mutex mutex_;
-  mutable std::shared_ptr<QLRowBlock> cache_;
-  mutable int cached_tablets_version_ = -1;
-  mutable int cached_tablet_locations_version_ = -1;
+  mutable std::shared_ptr<QLRowBlock> cache_ GUARDED_BY(mutex_);
+  mutable int cached_tablets_version_ GUARDED_BY(mutex_) = -1;
```

Comment:
This annotation and the one on `cached_tablet_locations_version_` are both good additions, but unfortunately lead to some compiler errors under certain build circumstances. This is caused by a failure of some compilers to recognize `std::shared_lock` as fulfilling the mutex-hold requirement.

Fortunately, there's an easy fix for this. Yugabyte has a custom `SharedLock` implementation (src/yb/util/shared_lock.h) with a workaround for the compiler issue. If you replace all instances of `std::shared_lock` in `yql_partitions_vtable.cc` with `SharedLock`, that should resolve the issue.

Reply:
Fixed. Thanks for the instructions!